resource_customizations/argoproj.io/ApplicationSet/health.lua

```diff
@@ -0,0 +1,22 @@
+hs = {}
+
+if obj.status ~= nil then
+  if obj.status.conditions ~= nil then
+    for i, condition in pairs(obj.status.conditions) do
+      if condition.type == "ErrorOccurred" and condition.status == "True" then
+        hs.status = "Degraded"
+        hs.message = condition.message
+        return hs
+      end
+      if condition.type == "ResourcesUpToDate" and condition.status == "True" then
+        hs.status = "Healthy"
+        hs.message = condition.message
+        return hs
+      end
+    end
+  end
+end
+
+hs.status = "Healthy"
```

Comment:
Could you add a comment stating that conditions were introduced in ApplicationSet v0.3. To support v0.2 and below we currentlyreturn Healthy, but will change to Progressing in the future.